File: Python/Programmers/문자열내마음대로정렬하기.py
def solution(strings, n):
    strings.sort()
    arr = []
    answer = []

    for i in range(len(strings)):
        arr.append([strings[i][n], i])
    arr.sort()

    for i in arr:
        answer.append(strings[i[1]])
    return answer
    # return sorted(strings, key=lambda x:x[n])

print(solution(["sun", "bed", "car"],1))
print(solution(["abce", "abcd", "cdx"],2))

File: Java/Programmers/문자열내마음대로정렬하기.java
import java.util.Arrays;

public class 문자열내마음대로정렬하기 {
	public static void main(String args[]) {
		Solution문자열내마음대로정렬하기 s = new Solution문자열내마음대로정렬하기();
		String[] strings = {"abce", "abcd", "cdx"};
		String[] res = s.solution(strings, 1);
		for(int i=0; i<res.length; i++) {
			System.out.print(res[i] + ", ");
		}
	}
}

class Solution문자열내마음대로정렬하기 {
	public String[] solution(String[] strings, int n) {
		String[] answer = new String[strings.length];
		for(int i=0; i<strings.length; i++) {
			answer[i] = String.valueOf(strings[i].charAt(n))+strings[i];
		}
		Arrays.sort(answer);
		
		for(int i=0; i<answer.length; i++) {
			answer[i] = answer[i].substring(1, answer[i].length());
		}
		return answer;
	}
}